internal/controllers/ipaddressclaim.go

```diff
@@ -366,17 +378,11 @@ func (r *IPAddressClaimReconciler) clusterToIPClaims(a client.Object) []reconcil
 	return requests
 }
 
-func buildAddressList(addressesInUse []ipamv1.IPAddress, gateway string) []string {
-	// Add extra capacity for the case that the pool's gateway is specified
-	addrStrings := make([]string, len(addressesInUse), len(addressesInUse)+1)
+func buildAddressList(addressesInUse []ipamv1.IPAddress) []string {
+	addrStrings := make([]string, len(addressesInUse))
 	for i, address := range addressesInUse {
 		addrStrings[i] = address.Spec.Address
 	}
-
-	if gateway != "" {
-		addrStrings = append(addrStrings, gateway)
-	}
-
```

Comment:
We no longer consider the gateway in use, but with this change set the gateway is not in available list.